pyomo/contrib/appsi/solvers/highs.py

```diff
@@ -221,6 +221,29 @@ def symbol_map(self):
         return SymbolMap()
```

Comment:
If you want this to be able to be used as a context manager, make sure to add an `__enter__` and `__exit__`, as in here: https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/solver/base.py#L73

This is not inherently supported in the APPSI base class.

Reply:
Understood, thank you. I just saw that you have a PR up that revamps a lot of the solver interfaces so I will do the least amount of changes and use it as a context manager in my test case

Reply:
The solver refactor PR is the next in what is probably going to be a fairly long series of changes, so good call 👍 That being said, the refactor is based on some fundamental design choices from APPSI so anything you do here should be portable without too much extra effort.

Reply:
Oh, I missed this conversation. Sorry.